### PR TITLE
feat: cria seção administrativa do frontend

### PR DESCRIPTION
**Pull request by FabioCarlesso**
Adiciona o módulo Admin em /admin (hub) e rotas placeholder
/admin/usuarios, /admin/usuarios/novo e /admin/usuarios/:id/editar,
todas protegidas por roleGuard(['ADMIN']).

- Cria componentes standalone AdminHomeComponent, UsuarioListComponent
  e UsuarioFormComponent seguindo o padrão do projeto.
- Inclui link "Administração" na navbar, exibido apenas para ADMIN.
- Adiciona testes unitários para os componentes e atualiza spec de
  rotas e do AppComponent para cobrir o novo link.
- Atualiza README.md e docs/context.md descrevendo a nova seção.

Refs #71

### CODE REVIEW

File: src/app/app.routes.ts [hunk 1]
```diff
@@ -182,6 +182,30 @@ export const routes: Routes = [
...
+  },
+  {
+    path: 'admin/usuarios/:id/editar',
+    canActivate: [roleGuard(['ADMIN'])],
+    loadComponent: () =>
+      import('./pages/admin/usuarios/usuario-form/usuario-form.component').then(m => m.UsuarioFormComponent)
+  },
   {
     path: 'profissionais',
     canActivate: [roleGuard(['ADMIN'])],
```
Comment (FabioCarlesso) [inline src/app/app.routes.ts:213]: **Angular Best Practices — Repeated guards on flat routes**

- **Problem:** Each of the four admin routes duplicates `canActivate: [roleGuard(['ADMIN'])]`. The routes are defined as siblings instead of using a parent route with `children`.
- **Why it matters:** If a fifth admin route is added, the guard can be accidentally omitted. A parent route with `canActivate` covers all children automatically, reducing the risk of an unprotected route.
- **Suggested improvement:** Group the admin routes under a single parent with `children`. Because the parent itself needs to render `AdminHomeComponent` at `''`, the structure becomes:
```ts
{
  path: 'admin',
  canActivate: [roleGuard(['ADMIN'])],
  children: [
    {
      path: '',
      loadComponent: () =>
        import('./pages/admin/admin-home/admin-home.component').then(m => m.AdminHomeComponent)
    },
    {
      path: 'usuarios',
      loadComponent: () =>
        import('./pages/admin/usuarios/usuario-list/usuario-list.component').then(m => m.UsuarioListComponent)
    },
    {
      path: 'usuarios/novo',
      loadComponent: () =>
        import('./pages/admin/usuarios/usuario-form/usuario-form.component').then(m => m.UsuarioFormComponent)
    },
    {
      path: 'usuarios/:id/editar',
      loadComponent: () =>
        import('./pages/admin/usuarios/usuario-form/usuario-form.component').then(m => m.UsuarioFormComponent)
    }
  ]
}
```
- **Severity:** Medium

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/app/pages/admin/usuarios/usuario-form/usuario-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  imports: [RouterLink],
+  templateUrl: './usuario-form.component.html',
+  styleUrl: './usuario-form.component.scss'
+})
+export class UsuarioFormComponent implements OnInit {
+  isEdit = false;
+  usuarioId: number | null = null;
+  parametroInvalido = false;
+
+  constructor(private route: ActivatedRoute) {}
```
Comment (FabioCarlesso) [inline src/app/pages/admin/usuarios/usuario-form/usuario-form.component.ts]: **Angular Best Practices — Constructor injection vs `inject()` function**

- **Problem:** `UsuarioFormComponent` uses constructor-based DI (`constructor(private route: ActivatedRoute) {}`), while the rest of the codebase (e.g., `roleGuard`) already uses the modern `inject()` function pattern.
- **Why it matters:** Mixing both styles in the same project creates inconsistency. The Angular team recommends `inject()` for standalone components and functional-style code.
- **Suggested improvement:**
```ts
export class UsuarioFormComponent implements OnInit {
  private readonly route = inject(ActivatedRoute);
  // ...
}
```
- **Severity:** Low

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/app/pages/admin/usuarios/usuario-form/usuario-form.component.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  ngOnInit(): void {
+    const rawId = this.route.snapshot.paramMap.get('id');
+    if (rawId === null) return;
+
```
Comment (FabioCarlesso) [inline src/app/pages/admin/usuarios/usuario-form/usuario-form.component.ts:21]: **Code Quality — `paramMap.get('id')` called twice**

- **Problem:** `this.route.snapshot.paramMap.get('id')` is called once manually to check for `null`, and then called again internally by `parseRouteNumberParam`. The `rawId` variable exists solely to guard what the utility already handles.
- **Why it matters:** Minor redundancy — if the key name changes, there are two places to update instead of one.
- **Suggested improvement:** Use `paramMap.has()` for the early-return guard to make the intent clear without repeating the key lookup:
```ts
ngOnInit(): void {
  if (!this.route.snapshot.paramMap.has('id')) return;

  this.usuarioId = parseRouteNumberParam(this.route.snapshot.paramMap, 'id');
  if (this.usuarioId === null) {
    this.parametroInvalido = true;
    return;
  }

  this.isEdit = true;
}
```
- **Severity:** Low

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/app/pages/admin/admin-home/admin-home.component.scss [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  padding: 1rem;
+  background: var(--white);
+  border: 1px solid var(--border);
+  border-radius: var(--radius);
+  color: var(--text);
+  text-decoration: none;
+  box-shadow: var(--shadow-1);
+
+  &:hover {
+    border-color: var(--primary);
```
Comment (FabioCarlesso) [inline src/app/pages/admin/admin-home/admin-home.component.scss:27]: **UI/UX — Missing `:focus-visible` style on `.admin-card`**

- **Problem:** The `.admin-card` anchor element only defines a `:hover` state. Keyboard users navigating with Tab will see no visible focus indicator, making the card inaccessible without a mouse.
- **Why it matters:** WCAG 2.1 Success Criterion 2.4.7 (Focus Visible) requires interactive elements to have a visible focus indicator. This is especially important for an admin panel.
- **Suggested improvement:**
```scss
.admin-card {
  // ... existing styles

  &:hover,
  &:focus-visible {
    border-color: var(--primary);
    outline: 2px solid var(--primary);
    outline-offset: 2px;
  }
}
```
- **Severity:** Medium

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/app/pages/admin/usuarios/usuario-list/usuario-list.component.html [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<div class="page-header">
+  <h1>Usuários</h1>
+  <a routerLink="/admin/usuarios/novo" class="btn btn-primary">+ Novo Usuário</a>
```
Comment (FabioCarlesso) [inline src/app/pages/admin/usuarios/usuario-list/usuario-list.component.html]: **UI/UX — Active "Novo Usuário" button on a placeholder page**

- **Problem:** The `+ Novo Usuário` button is fully clickable even though the user management feature is not yet implemented. Clicking it navigates to `UsuarioFormComponent`, which also shows a placeholder message.
- **Why it matters:** Offering an actionable button that leads to a dead-end degrades the user experience and may confuse administrators who expect to create a user.
- **Suggested improvement:** Either disable the button (`disabled` attribute / `btn--disabled` class) or remove it entirely until the feature is ready. Alternatively, a tooltip explaining the pending state would help.
```html
<button class="btn btn-primary" disabled title="Em breve">+ Novo Usuário</button>
```
- **Severity:** Low

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/app/app.routes.spec.ts [hunk 6]
```diff
@@ -92,4 +96,23 @@ describe('app routes', () => {
...
+    const listIndex = paths.indexOf('admin/usuarios');
+    const novoIndex = paths.indexOf('admin/usuarios/novo');
+    const editarIndex = paths.indexOf('admin/usuarios/:id/editar');
+
+    expect(homeIndex).toBeLessThan(listIndex);
+    expect(listIndex).toBeLessThan(novoIndex);
+    expect(novoIndex).toBeLessThan(editarIndex);
+  });
+
+  it('should protect every admin route with a canActivate guard', () => {
```
Comment (FabioCarlesso) [inline src/app/app.routes.spec.ts]: **Tests — Hardcoded route count is fragile**

- **Problem:** `expect(adminRoutes.length).toBe(4)` will silently break (unexpected test failure) whenever a new admin route is added, even if that route is correctly guarded.
- **Why it matters:** The purpose of this test is to ensure every admin route has a guard — but the count assertion adds a hidden coupling to the total number of routes, not to the guard coverage.
- **Suggested improvement:** Remove the count assertion and let the loop carry the intent:
```ts
it('should protect every admin route with a canActivate guard', () => {
  const adminRoutes = routes.filter(r => r.path?.startsWith('admin'));
  expect(adminRoutes.length).toBeGreaterThan(0);
  for (const route of adminRoutes) {
    expect(route.canActivate?.length).toBeGreaterThan(0);
  }
});
```
- **Severity:** Low

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/app/pages/admin/admin-home/admin-home.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { RouterTestingModule } from '@angular/router/testing';
```
Comment (FabioCarlesso) [inline src/app/pages/admin/admin-home/admin-home.component.spec.ts]: **Angular Best Practices — Deprecated `RouterTestingModule`**

- **Problem:** `RouterTestingModule` was deprecated in Angular 16 in favor of `provideRouter()` combined with `RouterTestingHarness` or `RouterModule`. The same pattern applies to `usuario-list.component.spec.ts` and `usuario-form.component.spec.ts`.
- **Why it matters:** Using deprecated APIs accumulates tech debt and may produce warnings in newer Angular versions. The project already targets Angular 19, so the modern API is fully available.
- **Suggested improvement:** Replace `RouterTestingModule` with `provideRouter([])` in the `TestBed` configuration:
```ts
import { provideRouter } from '@angular/router';

await TestBed.configureTestingModule({
  imports: [AdminHomeComponent],
  providers: [provideRouter([])]
}).compileComponents();
```
The same change should be applied to `usuario-list.component.spec.ts` and `usuario-form.component.spec.ts`.
- **Severity:** Medium

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/app/pages/admin/usuarios/usuario-form/usuario-form.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+  it('should render link back to user list', async () => {
+    await configure(null);
+    fixture = TestBed.createComponent(UsuarioFormComponent);
+    fixture.detectChanges();
+
+    const link = (fixture.nativeElement as HTMLElement).querySelector('a[href="/admin/usuarios"]');
+    expect(link).toBeTruthy();
+  });
+});
```
Comment (FabioCarlesso) [inline src/app/pages/admin/usuarios/usuario-form/usuario-form.component.spec.ts:82]: **Tests — Missing edge cases for `parseRouteNumberParam`**

- **Problem:** The spec covers `null`, a valid integer (`'42'`), and a non-numeric string (`'abc'`), but misses boundary values that `parseRouteNumberParam` is specifically designed to reject.
- **Why it matters:** The utility uses `Number.isSafeInteger` and a regex anchored at `[1-9]`, so `'0'`, `'-1'`, `'1.5'`, and `'9007199254740992'` (> `MAX_SAFE_INTEGER`) should all produce `parametroInvalido = true`. Without tests for these cases, a future refactor of the regex could silently regress.
- **Suggested improvement:** Add parametrized cases:
```ts
it.each(['0', '-1', '1.5', '9007199254740992'])(
  'should flag "%s" as invalid',
  async (id) => {
    await configure(id);
    fixture = TestBed.createComponent(UsuarioFormComponent);
    fixture.detectChanges();
    expect(fixture.componentInstance.parametroInvalido).toBeTrue();
  }
);
```
- **Severity:** Low

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/app/pages/admin/usuarios/usuario-list/usuario-list.component.scss
Comment (FabioCarlesso) [inline src/app/pages/admin/usuarios/usuario-list/usuario-list.component.scss:1]: **Code Quality — Duplicated `.empty-state` rule**

- **Problem:** `.empty-state { margin-bottom: 1.5rem; }` is identical in both `usuario-list.component.scss` and `usuario-form.component.scss`.
- **Why it matters:** If the design token or spacing value changes, both files need to be updated. Since `.empty-state` is already used in other pages in the project, it is likely a shared pattern.
- **Suggested improvement:** Move the rule to the global stylesheet (e.g., `src/styles.scss` or the appropriate shared partial) so all empty-state placeholders across the app are consistent and maintained in one place.
- **Severity:** Low

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by FabioCarlesso — COMMENTED** [review]
# Review — feat: cria seção administrativa do frontend

Overall this is a well-structured PR. The RBAC scaffolding is solid, documentation is thorough, and every new component comes with tests. The findings below are mostly low-to-medium concerns; none of them block the feature from working correctly in the current placeholder state.

---

## 1. Documentation ✅
`README.md` and `docs/context.md` are fully aligned with the changes: folder tree, route table, feature table, and the running summary paragraph are all updated. The "placeholder" nature of the sub-routes is clearly communicated. No issues found.

---

## 2. Bugs and Edge Cases ✅
No functional bugs found. The `parseRouteNumberParam` utility correctly guards the `id` parameter, and the `ngOnInit` logic properly distinguishes between "no param" (new mode) and "invalid param" (error state). The inline comment on `usuario-form.component.ts` flags a minor code-quality redundancy, not a bug.

---

## 3. Security ✅
All admin routes are protected by `roleGuard(['ADMIN'])`. The navbar link is conditionally rendered with `@if (authService.isAdmin())`. No unsafe HTML rendering, no exposed secrets, no insecure token handling. The guard implementation (`role.guard.ts`) follows the correct order: authentication check → role validation → session cleanup on corrupt state → redirect to `/403`. No issues found.

---

## 4. Performance ✅
All three components are lazy-loaded via `loadComponent`. Since these are pure presentational placeholders with no subscriptions, intervals, or API calls, there are no memory-leak or change-detection concerns. No issues found.

---

## 5. Angular Best Practices ⚠️
- **Medium** — Flat routes with repeated `canActivate` guards (see inline comment on `app.routes.ts`). A parent-with-children structure would centralise the guard and prevent accidental omissions on future sub-routes.
- **Medium** — `RouterTestingModule` is deprecated since Angular 16; the project targets Angular 19 (see inline comment on spec files).
- **Low** — Constructor injection in `UsuarioFormComponent` is inconsistent with the `inject()` style used elsewhere, e.g., in `roleGuard` (see inline comment).

---

## 6. API Integration ✅
No API calls in this PR — all components are placeholders by design. The documentation clearly states that the CRUD contracts will be implemented in future issues. No issues found.

---

## 7. UI/UX ⚠️
- **Medium** — `.admin-card` lacks a `:focus-visible` style, making it inaccessible for keyboard navigation (see inline comment on the SCSS file).
- **Low** — The "Novo Usuário" button on the placeholder list page leads to another placeholder, creating a dead-end interaction (see inline comment on the HTML file).

---

## 8. Tests ⚠️
Coverage is good for a placeholder PR — all three components and the app routes spec are updated. Two improvements noted:
- **Medium** — Deprecated `RouterTestingModule` should be replaced with `provideRouter([])` (see inline comment).
- **Low** — Hardcoded `.toBe(4)` count in `app.routes.spec.ts` will fail when any new admin route is added, even if it is correctly guarded (see inline comment).
- **Low** — `UsuarioFormComponent` spec is missing boundary-value cases for `parseRouteNumberParam` (`'0'`, `'-1'`, `'1.5'`, integers beyond `MAX_SAFE_INTEGER`) (see inline comment).

---

## 9. Code Quality ⚠️
- **Low** — `paramMap.get('id')` is called twice in `ngOnInit`; using `paramMap.has('id')` for the early-return guard removes the redundancy (see inline comment).
- **Low** — The `.empty-state` CSS rule is duplicated across `usuario-list` and `usuario-form` SCSS files; it belongs in a shared global partial (see inline comment).

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by FabioCarlesso** [comment]
## Correções aplicadas (commit `1963bac`)

Todas as 9 observações da revisão foram endereçadas neste commit. Resumo do que foi feito:

### Angular Best Practices
- **Rotas admin** — refatoradas de 4 rotas planas com `canActivate` repetido para uma única rota pai com `children`. O guard `roleGuard(['ADMIN'])` agora fica exclusivamente no pai e cobre todos os filhos automaticamente (`app.routes.ts`).
- **`inject()`** — `UsuarioFormComponent` trocou a injeção por construtor por `inject(ActivatedRoute)`, alinhando ao padrão já adotado no `roleGuard` e demais serviços do projeto.
- **`RouterTestingModule`** — substituído por `provideRouter([])` nos três specs (`admin-home`, `usuario-list`, `usuario-form`), eliminando a API depreciada desde Angular 16.

### Code Quality
- **`paramMap.has()`** — o `ngOnInit` de `UsuarioFormComponent` passou a usar `paramMap.has('id')` para o early-return, removendo a leitura dupla do parâmetro de rota.
- **`.empty-state` duplicado** — regra removida de `usuario-list.component.scss` e `usuario-form.component.scss`; a definição já existente em `src/styles.scss` é suficiente.

### UI/UX & Acessibilidade
- **`:focus-visible`** — adicionado ao `.admin-card` em `admin-home.component.scss`, garantindo indicador de foco visível para usuários de teclado.
- **Botão "Novo Usuário"** — o `<a routerLink>` foi substituído por `<button disabled>` na tela placeholder de `usuario-list`, evitando que o administrador navegue para um formulário ainda não implementado.

### Tests
- **Contagem hardcoded** — o `expect(adminRoutes.length).toBe(4)` foi removido; o teste foi reescrito para verificar que a rota pai `admin` possui `canActivate` e `children`, refletindo a nova estrutura.
- **`getAllPaths` helper** — adicionado ao `app.routes.spec.ts` para achatar rotas com `children` e manter o teste `should contain all expected routes` funcionando após a refatoração.
- **Boundary values** — adicionados 4 novos casos em `usuario-form.component.spec.ts` cobrindo `'0'`, `'-1'`, `'1.5'` e `'9007199254740992'` (> `Number.MAX_SAFE_INTEGER`), garantindo cobertura das rejeições do `parseRouteNumberParam`.

---
_Generated by [Claude Code](https://claude.ai/code)_
